qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/WarmupTestsIT.java

```diff
@@ -0,0 +1,99 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.bwc;
+
+import org.opensearch.knn.index.SpaceType;
+
+import java.util.Collections;
+
+import static org.opensearch.knn.TestUtils.KNN_ALGO_PARAM_M_MIN_VALUE;
+import static org.opensearch.knn.TestUtils.KNN_ALGO_PARAM_EF_CONSTRUCTION_MIN_VALUE;
+import static org.opensearch.knn.TestUtils.KNN_ENGINE_FAISS;
+import static org.opensearch.knn.TestUtils.NODES_BWC_CLUSTER;
+
+public class WarmupTestsIT extends AbstractRestartUpgradeTestCase {
+    private static final String TEST_FIELD = "test-field";
+    private static final int DIMENSIONS = 5;
+    private static final int K = 5;
+    private static final int ADD_DOCS_CNT = 10;
+
+    // Default Legacy Field Mapping
+    // space_type : "l2", engine : "nmslib", m : 16, ef_construction : 512
+    public void testKnnWarmupDefaultLegacyFieldMapping() throws Exception {
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+
+        if (isRunningAgainstOldCluster()) {
+            createKnnIndex(testIndex, getKNNDefaultIndexSettings(), createKnnIndexMapping(TEST_FIELD, DIMENSIONS));
+            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 0, ADD_DOCS_CNT);
+        }
+
+        else {
+            kNNWarmupUpgradedCluster();
+        }
+    }
+
+    // Custom Legacy Field Mapping
+    // space_type : "linf", engine : "nmslib", m : 2, ef_construction : 2
+    public void testKnnWarmupCustomLegacyFieldMapping() throws Exception {
+        if (isRunningAgainstOldCluster()) {
+            createKnnIndex(
+                testIndex,
+                getKNNIndexCustomLegacyFieldMappingSettings(
+                    SpaceType.LINF,
+                    KNN_ALGO_PARAM_M_MIN_VALUE,
+                    KNN_ALGO_PARAM_EF_CONSTRUCTION_MIN_VALUE
+                ),
+                createKnnIndexMapping(TEST_FIELD, DIMENSIONS)
```

Comment:
IMO it is hard to read in one or two places where spotless refactored it into multiple lines. If we refactor createIndex everywhere like you suggested, it will increase the lines of code like shown below:
```
createKnnIndex(testIndex, getKNNDefaultIndexSettings(), createKnnIndexMapping(TEST_FIELD, DIMENSIONS));
```

```
Settings defaultIndexSettings = getKNNDefaultIndexSettings();
String indexMapping = createKnnIndexMapping(TEST_FIELD, DIMENSIONS);
createKnnIndex(testIndex, defaultIndexSettings, indexMapping);
```

Reply:
I had in mind only this place, change will look something like one below. IMO more lines are ok, just because of the multiple braces/scopes it's easy to mix them and make a mistake while changing it in future. 

```
createKnnIndex(
                testIndex,
                getKNNIndexCustomLegacyFieldMappingSettings(
                    SpaceType.LINF,
                    KNN_ALGO_PARAM_M_MIN_VALUE,
                    KNN_ALGO_PARAM_EF_CONSTRUCTION_MIN_VALUE
                ),
                createKnnIndexMapping(TEST_FIELD, DIMENSIONS)
 );
```
I suggest we change it to 

```
String indexMappingSettings = getKNNIndexCustomLegacyFieldMappingSettings(
                    SpaceType.LINF,
                    KNN_ALGO_PARAM_M_MIN_VALUE,
                    KNN_ALGO_PARAM_EF_CONSTRUCTION_MIN_VALUE);
String indexMapping = createKnnIndexMapping(TEST_FIELD, DIMENSIONS);
createKnnIndex(testIndex, indexMappingSettings, indexMapping);
```

Reply:
Got it. If it is just this one, it is good to change it. Will update it.